### PR TITLE
test: Functional and Logical Tests for ERPNext Delivery Note Trends Report

### PR DESCRIPTION
**Pull request by hrishikesh8848**
### Title: 
Add Test Cases for Delivery Note Trends Report

### Test Summary
This test suite ensures functional and logical correctness of the Delivery Note Trends Report in ERPNext. It verifies that the report responds correctly to valid filters, handles various chart scenarios, and gracefully processes empty or edge input.

### Scenarios Covered
Input Validations
T_DNT_001 – Report Execution with Valid Filters
Validates that the report returns expected columns, data, and chart data when provided with proper filters like company, date range, fiscal year, and grouping.

T_DNT_002 – Execution Using Fiscal Year Filter
Checks that the report can be executed with only a fiscal year and grouping parameters, and still returns valid outputs.

T_DNT_004 – Execution with Minimal Required Filters
Similar to T_DNT_002, it tests another variant of minimal input, ensuring the report can still compute successfully.

Edge Cases
T_DNT_003 – Empty Chart Data (Empty Report)
Verifies that get_chart_data returns an empty list when the input data is empty.

T_DNT_005 – Duplicate of T_DNT_003 (Redundant, but Safe)
Serves as a redundancy to validate that empty data is consistently handled in get_chart_data.

T_DNT_006 – Chart with Grouped Data
Simulates data with group labels and ensures the bar chart correctly groups and represents them with proper labels and currency type.

T_DNT_007 – Chart with More Than 10 Records
Ensures the chart data logic limits the output to top 10 labels when more than 10 records are present.

T_DNT_008 – Chart with Less Than 10 Non-grouped Entries
Verifies the chart shows all available entries when less than 10 records exist and no grouping is applied.

### Technology
Test Framework: unittest

Test Base Class: unittest.TestCase

### Linked Feature/Bug
None

### CODE REVIEW

File: erpnext/stock/report/delivery_note_trends/test_delivery_note_trends.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+			fiscal_year.year = "2024-2025"
+			fiscal_year.year_start_date = "2024-04-01"
+			fiscal_year.year_end_date = "2025-03-31"
+			fiscal_year.companies[0].company = "_Test Company"
+			fiscal_year.save()
+		else:
+			fiscal_year = frappe.get_doc("Fiscal Year", "2024-2025")
+			if not any(d.company == "_Test Company" for d in fiscal_year.companies):
+				fiscal_year.append("companies", {"company": "_Test Company"})
+				fiscal_year.save()
```
Comment (aasif-patel) [inline erpnext/stock/report/delivery_note_trends/test_delivery_note_trends.py]: there is method get_or_create_fiscal_year which handles this situation better

**Review by aasif-patel — CHANGES_REQUESTED** [review]

File: erpnext/stock/report/delivery_note_trends/test_delivery_note_trends.py (hunk 1 above)
Comment (hrishikesh8848) [reply erpnext/stock/report/delivery_note_trends/test_delivery_note_trends.py]: okay

Comment (hrishikesh8848) [reply erpnext/stock/report/delivery_note_trends/test_delivery_note_trends.py]: @aasif-patel this is done

**Review by aasif-patel — APPROVED** [review]

**Comment by swathibollineni2** [comment]
@hrishikesh8848 All test cases are failed created an issue https://github.com/8848digital/erpnext/issues/2575
